libs/brig-types/src/Brig/Types/User.hs

```diff
@@ -273,36 +279,100 @@ instance ToJSON SelfProfile where
 
 ----------------------------------------------------------------------------
 -- Rich info
+data RichInfo = RichInfo { richInfoMap :: Map (CI Text) Text
+                         , richInfoAssocList :: [RichField]
+                         }
+  deriving (Eq, Show, Generic)
 
-data RichInfo = RichInfo
-    { richInfoFields :: ![RichField]  -- ^ An ordered list of fields
-    }
-    deriving (Eq, Show, Generic)
+newtype RichInfoAssocList = RichInfoAssocList [RichField]
+  deriving (Eq, Show, Generic)
 
-instance ToJSON RichInfo where
-    toJSON u = object
-        [ "fields" .= richInfoFields u
-        , "version" .= (0 :: Int)
-        ]
+toRichInfoAssocList :: RichInfo -> RichInfoAssocList
+toRichInfoAssocList (RichInfo mp al) = RichInfoAssocList . nubrf $ toal mp <> al
+  where
+    nubrf :: [RichField] -> [RichField]
+    nubrf = nubBy ((==) `on` \(RichField k _) -> CI.mk k)
+
+    toal :: Map (CI Text) Text -> [RichField]
+    toal = map (\(k, v) -> RichField k v) . Map.toAscList
+
+richInfoMapURN, richInfoAssocListURN :: Text
+richInfoMapURN = "urn:ietf:params:scim:schemas:extension:wire:1.0:User"
```

Comment:
There was a TODO whether `2.0` was needed in the string. It is not. so this is fine :)